uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-balance-tab.component.js

```diff
@@ -87,20 +58,39 @@ define([
      * sorted by title in an ascending order.
      */
     function loadAbsenceTypes () {
-      return AbsenceType.all({options: { sort: 'title ASC' }})
+      return AbsenceType.all({ options: { sort: 'title ASC' } })
       .then(function (response) {
         vm.absenceTypes = response;
       });
     }
 
     /**
+     * Loads all dependencies needed by the component and its children.
+     *
+     * @return {Promise}
+     */
+    function loadDependencies () {
+      return $q.all([
+        loadAbsencePeriods(),
+        loadAbsenceTypes(),
+        loadLoggedInContactId()
+      ])
+      .catch(function (error) {
+        notification.error('Error', error);
+      })
+      .finally(function () {
+        vm.loading.component = false;
```

Comment:
i wonder if this would be better out, of `loadDependencies`, directly in the `init` function
```js
(function init () {
  initWatchers();

  loadDependencies.then(function () {
    vm.loading.component;
  });
}());
```
i think this reads better, and makes sure that `loadDependencies` does only one thing, instead of two